### ISSUE TITLE
missing libs: launcher_client.so, libvstdlib_client.so, libsteam_api.so and scaleformui_client.so

### ISSUE THREAD

**Issue by oknyshuk** [tyabus/Kisak-Strike#5]
after building with `cmake .. -DUSE_ROCKETUI=1 -DDEDICATED=0; make -j20` I am greeted with 
```
Failed to load the launcher(bin/linux64/launcher_client.so) (bin/linux64/launcher_client.so: cannot open shared object file: No such file or directory)
```
on the first launch, after copying the lib from depot 734, this is the netx message:
```
Failed to load the launcher(bin/linux64/launcher_client.so) (libvstdlib_client.so: cannot open shared object file: No such file or directory)
```
then, after copying libvstdlib_client.so from depot 734, I have this error:
```
Failed to load the launcher(bin/linux64/launcher_client.so) (libsteam_api.so: cannot open shared object file: No such file or directory)
```
only after I copy libsteam_api.so from goldberg's steam emu to bin/linux64 and have steam launched in the background, I can launch the game, but before getting to any gui, like LWSS's original KS does, I have this:
```
$ ./csgo_linux64
SDL video target is 'wayland'
This system supports the OpenGL extension GL_EXT_framebuffer_object.
This system supports the OpenGL extension GL_EXT_framebuffer_blit.
This system supports the OpenGL extension GL_EXT_framebuffer_multisample.
zsh: segmentation fault (core dumped)  ./csgo_linux64

$ SDL_VIDEODRIVER=x11 ./csgo_linux64 -console
SDL video target is 'x11'
This system supports the OpenGL extension GL_EXT_framebuffer_object.
This system supports the OpenGL extension GL_EXT_framebuffer_blit.
This system supports the OpenGL extension GL_EXT_framebuffer_multisample.
This system DOES NOT support the OpenGL extension GL_APPLE_fence.
This system DOES NOT support the OpenGL extension GL_NV_fence.
This system supports the OpenGL extension GL_ARB_sync.
This system supports the OpenGL extension GL_EXT_draw_buffers2.
This system DOES NOT support the OpenGL extension GL_EXT_bindable_uniform.
This system DOES NOT support the OpenGL extension GL_APPLE_flush_buffer_range.
This system supports the OpenGL extension GL_ARB_map_buffer_range.
This system supports the OpenGL extension GL_ARB_vertex_buffer_object.
This system supports the OpenGL extension GL_ARB_occlusion_query.
This system DOES NOT support the OpenGL extension GL_APPLE_texture_range.
This system DOES NOT support the OpenGL extension GL_APPLE_client_storage.
This system DOES NOT support the OpenGL extension GL_ARB_uniform_buffer.
This system supports the OpenGL extension GL_ARB_vertex_array_bgra.
This system supports the OpenGL extension GL_EXT_vertex_array_bgra.
This system supports the OpenGL extension GL_ARB_framebuffer_object.
This system DOES NOT support the OpenGL extension GL_GREMEDY_string_marker.
This system supports the OpenGL extension GL_ARB_debug_output.
This system supports the OpenGL extension GL_EXT_direct_state_access.
This system DOES NOT support the OpenGL extension GL_NV_bindless_texture.
This system supports the OpenGL extension GL_AMD_pinned_memory.
This system supports the OpenGL extension GL_EXT_framebuffer_multisample_blit_scaled.
This system supports the OpenGL extension GL_EXT_texture_sRGB_decode.
This system DOES NOT support the OpenGL extension GL_NVX_gpu_memory_info.
This system DOES NOT support the OpenGL extension GL_ATI_meminfo.
This system supports the OpenGL extension GL_EXT_texture_compression_s3tc.
This system supports the OpenGL extension GL_EXT_texture_compression_dxt1.
This system supports the OpenGL extension GL_ANGLE_texture_compression_dxt3.
This system supports the OpenGL extension GL_ANGLE_texture_compression_dxt5.
This system supports the OpenGL extension GL_ARB_buffer_storage.
This system supports the OpenGL extension GLX_EXT_swap_control_tear.
Module scaleformui_client.so failed to load! Error: ((null))
AppFramework : Unable to load module scaleformui_client.so!
Unable to load interface ScaleformUI002 from scaleformui_client.so, requested from EXE.
```

I'm on Arch, using Gnome on Wayland. LWSS's KSGO worked, but crashed often. this fork doesn't launch at all. please, help me fix this.

regards, Oleksandr

**Comment by oknyshuk** [comment]
if I copy scaleformui_client.so from depot 734, I have this error:
```
failed to dlopen scaleformui_client.so error=scaleformui_client.so: wrong ELF class: ELFCLASS32
Module scaleformui_client.so failed to load! Error: ((null))
AppFramework : Unable to load module scaleformui_client.so!
Unable to load interface ScaleformUI002 from scaleformui_client.so, requested from EXE.
```
P.S.: at first I forgot to `chmod +x` the libs I copied from depot, but now I did that

**Comment by oknyshuk** [comment]
@tyabus? you there?

**Comment by tyabus** [comment]
> if I copy scaleformui_client.so from depot 734, I have this error:
> 
> ```
> failed to dlopen scaleformui_client.so error=scaleformui_client.so: wrong ELF class: ELFCLASS32
> Module scaleformui_client.so failed to load! Error: ((null))
> AppFramework : Unable to load module scaleformui_client.so!
> Unable to load interface ScaleformUI002 from scaleformui_client.so, requested from EXE.
> ```
> 
> P.S.: at first I forgot to `chmod +x` the libs I copied from depot, but now I did that

You're trying to load 32 bit library while using 64 bit engine.

**Comment by oknyshuk** [comment]
thanks, I didn't notice that I copied `scaleformui_client.so` from `depots/734/11602641/bin`, and not from `depots/734/11602641/bin/linux64`. why do I need it anyway? original repo didn't require proprietary library to launch

**Comment by oknyshuk** [comment]
I have a [segfault](https://pastebin.com/UZMAa6zn) after seeing a black window for 2 seconds:
```
#0  0x000055b61b724351 in RocketMainMenuDocument::LoadDialog () at /home/k1gen/development/csgo/Kisak-Strike/game/client/cstrike15/RocketUI/rkmenu_main.cpp:31
#1  0x000055b61b724575 in RocketMainMenuDocument::ShowPanel (bShow=<optimized out>, immediate=<optimized out>)
    at /home/k1gen/development/csgo/Kisak-Strike/game/client/cstrike15/RocketUI/rkmenu_main.cpp:76
#2  0x000055b64dda8b73 in CEngineVGui::ActivateGameUI (this=this@entry=0x55b64e792320 <g_EngineVGuiImp>)
    at /home/k1gen/development/csgo/Kisak-Strike/engine/vgui_baseui_interface.cpp:1347
#3  0x000055b64ddaa9fe in CEngineVGui::ActivateGameUI (this=0x55b64e792320 <g_EngineVGuiImp>)
    at /home/k1gen/development/csgo/Kisak-Strike/engine/vgui_baseui_interface.cpp:1327
#4  CEngineVGui::Init (this=0x55b64e792320 <g_EngineVGuiImp>) at /home/k1gen/development/csgo/Kisak-Strike/engine/vgui_baseui_interface.cpp:1050
#5  0x000055b64dc3d3d9 in Host_Init (bDedicated=false) at /home/k1gen/development/csgo/Kisak-Strike/engine/host.cpp:5666
#6  0x000055b64dcb3479 in Sys_InitGame (appSystemFactory=<optimized out>, pBaseDir=pBaseDir@entry=0x55b690257e20 "", pwnd=<optimized out>, 
    bIsDedicated=bIsDedicated@entry=0) at /home/k1gen/development/csgo/Kisak-Strike/engine/sys_dll.cpp:1152
#7  0x000055b64dcb895f in CEngine::Load (this=<optimized out>, dedicated=false, rootdir=0x55b690257e20 "")
    at /home/k1gen/development/csgo/Kisak-Strike/engine/sys_engine.cpp:245
#8  0x000055b64dcb585b in CModAppSystemGroup::Main (this=<optimized out>) at /home/k1gen/development/csgo/Kisak-Strike/engine/sys_dll2.cpp:2411
#9  0x000055b64df48139 in CAppSystemGroup::Run (this=this@entry=0x7ffc2ba658d0) at /home/k1gen/development/csgo/Kisak-Strike/appframework/AppSystemGroup.cpp:775
#10 0x000055b64dcb72b3 in CEngineAPI::RunListenServer (this=0x55b64e654600 <s_EngineAPI>) at /home/k1gen/development/csgo/Kisak-Strike/engine/sys_dll2.cpp:1437
#11 0x000055b690034349 in ?? () from bin/linux64/launcher_client.so
#12 0x000055b690034349 in ?? () from bin/linux64/launcher_client.so
#13 0x000055b69000ff99 in LauncherMain () from bin/linux64/launcher_client.so
#14 0x000055b690327850 in __libc_start_call_main (main=main@entry=0x55b6908bb080 <main(int, char**)>, argc=argc@entry=2, argv=argv@entry=0x7ffc2ba65dc8)
    at ../sysdeps/nptl/libc_start_call_main.h:58
#15 0x000055b69032790a in __libc_start_main_impl (main=0x55b6908bb080 <main(int, char**)>, argc=2, argv=0x7ffc2ba65dc8, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7ffc2ba65db8) at ../csu/libc-start.c:360
#16 0x000055b6908bb135 in _start ()
```
should I create another issue and close this one?

**Comment by tyabus** [comment]
> I have a [segfault](https://pastebin.com/UZMAa6zn) after seeing a black window for 2 seconds:
> 
> ```
> #0  0x000055b61b724351 in RocketMainMenuDocument::LoadDialog () at /home/k1gen/development/csgo/Kisak-Strike/game/client/cstrike15/RocketUI/rkmenu_main.cpp:31
> #1  0x000055b61b724575 in RocketMainMenuDocument::ShowPanel (bShow=<optimized out>, immediate=<optimized out>)
>     at /home/k1gen/development/csgo/Kisak-Strike/game/client/cstrike15/RocketUI/rkmenu_main.cpp:76
> #2  0x000055b64dda8b73 in CEngineVGui::ActivateGameUI (this=this@entry=0x55b64e792320 <g_EngineVGuiImp>)
>     at /home/k1gen/development/csgo/Kisak-Strike/engine/vgui_baseui_interface.cpp:1347
> #3  0x000055b64ddaa9fe in CEngineVGui::ActivateGameUI (this=0x55b64e792320 <g_EngineVGuiImp>)
>     at /home/k1gen/development/csgo/Kisak-Strike/engine/vgui_baseui_interface.cpp:1327
> #4  CEngineVGui::Init (this=0x55b64e792320 <g_EngineVGuiImp>) at /home/k1gen/development/csgo/Kisak-Strike/engine/vgui_baseui_interface.cpp:1050
> #5  0x000055b64dc3d3d9 in Host_Init (bDedicated=false) at /home/k1gen/development/csgo/Kisak-Strike/engine/host.cpp:5666
> #6  0x000055b64dcb3479 in Sys_InitGame (appSystemFactory=<optimized out>, pBaseDir=pBaseDir@entry=0x55b690257e20 "", pwnd=<optimized out>, 
>     bIsDedicated=bIsDedicated@entry=0) at /home/k1gen/development/csgo/Kisak-Strike/engine/sys_dll.cpp:1152
> #7  0x000055b64dcb895f in CEngine::Load (this=<optimized out>, dedicated=false, rootdir=0x55b690257e20 "")
>     at /home/k1gen/development/csgo/Kisak-Strike/engine/sys_engine.cpp:245
> #8  0x000055b64dcb585b in CModAppSystemGroup::Main (this=<optimized out>) at /home/k1gen/development/csgo/Kisak-Strike/engine/sys_dll2.cpp:2411
> #9  0x000055b64df48139 in CAppSystemGroup::Run (this=this@entry=0x7ffc2ba658d0) at /home/k1gen/development/csgo/Kisak-Strike/appframework/AppSystemGroup.cpp:775
> #10 0x000055b64dcb72b3 in CEngineAPI::RunListenServer (this=0x55b64e654600 <s_EngineAPI>) at /home/k1gen/development/csgo/Kisak-Strike/engine/sys_dll2.cpp:1437
> #11 0x000055b690034349 in ?? () from bin/linux64/launcher_client.so
> #12 0x000055b690034349 in ?? () from bin/linux64/launcher_client.so
> #13 0x000055b69000ff99 in LauncherMain () from bin/linux64/launcher_client.so
> #14 0x000055b690327850 in __libc_start_call_main (main=main@entry=0x55b6908bb080 <main(int, char**)>, argc=argc@entry=2, argv=argv@entry=0x7ffc2ba65dc8)
>     at ../sysdeps/nptl/libc_start_call_main.h:58
> #15 0x000055b69032790a in __libc_start_main_impl (main=0x55b6908bb080 <main(int, char**)>, argc=2, argv=0x7ffc2ba65dc8, init=<optimized out>, fini=<optimized out>, 
>     rtld_fini=<optimized out>, stack_end=0x7ffc2ba65db8) at ../csu/libc-start.c:360
> #16 0x000055b6908bb135 in _start ()
> ```
> 
> should I create another issue and close this one?

You either use scaleform or rocketui

**Comment by oknyshuk** [comment]
come on, you saw yourself that without scaleform lib ksgo doesn't launch. and I chose -DUSE_ROCKETUI=1 when compiling

**Comment by tyabus** [comment]
> come on, you saw yourself that without scaleform lib ksgo doesn't launch. and I chose -DUSE_ROCKETUI=1 when compiling

Do you have the required resources for rocketui?

**Comment by oknyshuk** [comment]
wdym "resources"?

**Comment by tyabus** [comment]
> wdym "resources"?

https://github.com/tyabus/Kisak-Strike-Files

**Comment by oknyshuk** [comment]
yeah, I copied them, but it dumps core again. here's the stdout:
```
SDL video target is 'x11'
This system supports the OpenGL extension GL_EXT_framebuffer_object.
This system supports the OpenGL extension GL_EXT_framebuffer_blit.
This system supports the OpenGL extension GL_EXT_framebuffer_multisample.
This system DOES NOT support the OpenGL extension GL_APPLE_fence.
This system DOES NOT support the OpenGL extension GL_NV_fence.
This system supports the OpenGL extension GL_ARB_sync.
This system supports the OpenGL extension GL_EXT_draw_buffers2.
This system DOES NOT support the OpenGL extension GL_EXT_bindable_uniform.
This system DOES NOT support the OpenGL extension GL_APPLE_flush_buffer_range.
This system supports the OpenGL extension GL_ARB_map_buffer_range.
This system supports the OpenGL extension GL_ARB_vertex_buffer_object.
This system supports the OpenGL extension GL_ARB_occlusion_query.
This system DOES NOT support the OpenGL extension GL_APPLE_texture_range.
This system DOES NOT support the OpenGL extension GL_APPLE_client_storage.
This system DOES NOT support the OpenGL extension GL_ARB_uniform_buffer.
This system supports the OpenGL extension GL_ARB_vertex_array_bgra.
This system supports the OpenGL extension GL_EXT_vertex_array_bgra.
This system supports the OpenGL extension GL_ARB_framebuffer_object.
This system DOES NOT support the OpenGL extension GL_GREMEDY_string_marker.
This system supports the OpenGL extension GL_ARB_debug_output.
This system supports the OpenGL extension GL_EXT_direct_state_access.
This system DOES NOT support the OpenGL extension GL_NV_bindless_texture.
This system supports the OpenGL extension GL_AMD_pinned_memory.
This system supports the OpenGL extension GL_EXT_framebuffer_multisample_blit_scaled.
This system supports the OpenGL extension GL_EXT_texture_sRGB_decode.
This system DOES NOT support the OpenGL extension GL_NVX_gpu_memory_info.
This system DOES NOT support the OpenGL extension GL_ATI_meminfo.
This system supports the OpenGL extension GL_EXT_texture_compression_s3tc.
This system supports the OpenGL extension GL_EXT_texture_compression_dxt1.
This system supports the OpenGL extension GL_ANGLE_texture_compression_dxt3.
This system supports the OpenGL extension GL_ANGLE_texture_compression_dxt5.
This system supports the OpenGL extension GL_ARB_buffer_storage.
This system supports the OpenGL extension GLX_EXT_swap_control_tear.
Using breakpad minidump system 730/13580.500
USRLOCAL path using Steam profile data folder:
/home/k1gen/.local/share/Goldberg SteamEmu Saves/730/local
Did not detect any valid joysticks.
RESZ NOT SUPPORTED!
INTZ NOT SUPPORTED!
RESZ NOT SUPPORTED!
INTZ NOT SUPPORTED!
Module /home/k1gen/development/csgo/game/bin/linux64/stdshader_dbg failed to load! Error: ((null))
Module stdshader_dbg failed to load! Error: ((null))
Failed to read the default inventory image file (materials/vgui/inventory_default.vtf)

 ##### swap interval = 0     swap limit = 1 #####
Module /home/k1gen/development/csgo/game/csgo/bin/matchmaking_client.so failed to load! Error: ((null))
Module /home/k1gen/development/csgo/game/csgo/bin/client_client.so failed to load! Error: ((null))
CClientSteamContext logged on = 1
Module /home/k1gen/development/csgo/game/csgo/bin/server failed to load! Error: ((null))
Game.dll loaded for "Counter-Strike: Global Offensive"
CGameEventManager::AddListener: event 'server_pre_shutdown' unknown.
CGameEventManager::AddListener: event 'game_newmap' unknown.
CGameEventManager::AddListener: event 'finale_start' unknown.
CGameEventManager::AddListener: event 'round_start' unknown.
CGameEventManager::AddListener: event 'round_end' unknown.
CGameEventManager::AddListener: event 'difficulty_changed' unknown.
CGameEventManager::AddListener: event 'player_connect' unknown.
CGameEventManager::AddListener: event 'player_disconnect' unknown.
GameTypes: missing mapgroupsSP entry for game type/mode (custom/custom).
GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/cooperative).
GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/coopmission).
zsh: segmentation fault (core dumped)  SDL_VIDEODRIVER=x11 ./csgo_linux64
```

**Comment by oknyshuk** [comment]
@tyabus?

**Comment by oknyshuk** [comment]
hey @tyabus, do you have any idea why it's segfaulting?

**Comment by oknyshuk** [comment]
still can't run your fork

**Comment by oknyshuk** [comment]
come on, you're keeping this project alive - but I can't use it. please, help me

**Comment by oknyshuk** [comment]
after restarting from scratch, I still need to copy two libs from the depot (launcher_client.so and libvstdlib_client.so) and use goldberg's libsteam_api.so, and with those libs in bin/linux64, I get `Failed to load the launcher(bin/linux64/launcher_client.so) (bin/linux64/launcher_client.so: undefined symbol: Plat_VerifyHardwareKeyPrompt)`. @tyabus, I need your help, please stop ignoring me

**Comment by oknyshuk** [comment]
@tyabus: I think this error is related to this: https://github.com/tyabus/Kisak-Strike/commit/43b43bd62ab5de7cf1640b576516105787bbda66#diff-8bf06743fa4ba3ee5ef5abcfe0b14c3d75eab403c49ca1661c0a1e2bf18319f0L97

**Comment by oknyshuk** [comment]
@tyabus?

**Comment by tyabus** [comment]
Closing because i cannot reproduce this.

**Comment by oknyshuk** [comment]
I can start from scratch once again, please provide commands you have used (especially what files did you copy and from which depots), so I can test if this issue is now gone

**Comment by yumirak** [comment]
this main issue is really just `launcher_client.so` is missing because complier output  `launcher.so` instead
so to fix it either rename "launcher.so" to "launcher_client.so" or edit 
`launcher/CMakelist.txt` 

> set(OUTBINNAME "launcher")

 to 

> set(OUTBINNAME "launcher_client")

and recompile.

**Comment by oknyshuk** [comment]
> this main issue is really just `launcher_client.so` is missing because complier output `launcher.so` instead so to fix it either rename "launcher.so" to "launcher_client.so" or edit `launcher/CMakelist.txt`
> 
> > set(OUTBINNAME "launcher")
> 
> to
> 
> > set(OUTBINNAME "launcher_client")
> 
> and recompile.

thanks, @yumirak, that did it! and sorry for pinging you so much, tyabus
